plotting_refactor/Dataset.py

```diff
@@ -0,0 +1,63 @@
+import time
+
+
+class Dataset:
+    def __init__(self, fitpage_index, x_data, y_data, y_fit, data_2d, plotpage_index=0):
+        self.fitpage_index = fitpage_index
+        self.x_data = x_data
+        self.y_data = y_data
+        self.y_fit = y_fit
+        self.plotpage_index = plotpage_index
```

Comment:
As discussed at the call today and tangentially mentioned in other comments, the data should have no knowledge of the plots, and the plots should be a view of the data. `plotpage_index` would be better served elsewhere.